code/modules/antagonists/wraith/abilties/trickster/possess.dm

```diff
@@ -1,108 +1,176 @@
-			APPLY_ATOM_PROPERTY(H, PROP_MOB_NO_SELF_HARM, H)	//Subject to change.
-			sleep(45 SECONDS)
-			if (!H?.loc)
-				target_player?.dnr--
-				return
-			boutput(H, "<span class='bold' style='color:red;font-size:150%'>Your control on this body is weakening, you will soon be kicked out of it.</span>")
-			sleep(20 SECONDS)
-			if(!H?.loc && !W.loc) //Everyone's dead, go home
-				target_player?.dnr--
-				return
-			if(!W.loc) //wraith got gibbed, kick them into the aether and put the human back
-				boutput(H, SPAN_ALERT("You are torn apart from the body you were in but cannot find your ethereal self! You are thrown into the otherworld as a powerless ghost."))
-				H.ghostize()
-				REMOVE_ATOM_PROPERTY(H, PROP_MOB_NO_SELF_HARM, H)
-				if (human_mind)
-					human_mind.transfer_to(H)
-					playsound(H, 'sound/effects/ghost2.ogg', 50, FALSE)
-					boutput(H, SPAN_NOTICE("You slowly regain control of your body. It's as if the presence within you dissipated into nothingness."))
-				target_player?.dnr--
-				return
-			target_player?.dnr--
-			if(!H?.loc) //Human gibbed, put the wraith back into their body
-				src.return_wraith(H)
-			else
-				boutput(H, "<span class='bold' style='color:red;font-size:150%'>Your hold on this body has been broken! You return to the aether.</span>")
-				REMOVE_ATOM_PROPERTY(H, PROP_MOB_NO_SELF_HARM, H)
-				src.return_wraith(H)
-				if (human_mind)
-					human_mind.transfer_to(H)
-					playsound(H, 'sound/effects/ghost2.ogg', 50, FALSE)
-			qdel(WG)
-			H.take_brain_damage(30)
-			H.setStatus("knockdown", 5 SECOND)
-			boutput(H, SPAN_NOTICE("The presence has left your body and you are thrusted back into it, immediately assaulted with a ringing headache."))
-		return FALSE
-
-	proc/return_wraith(mob/possessed) //we want to be absolutely sure the wraith goes back to their body no matter what
-		var/datum/abilityHolder/wraith/AH = src.holder
-		AH.possession_points = 0
-		UnregisterSignal(possessed, COMSIG_MOB_DEATH)
-		if (QDELETED(src.holder.owner)) //hopefully catch the wraith dying?
+				src.human_observer = O.insert_slasher_observer(T)
+		src.wraith_mind.transfer_to(T)
+		src.wraith_mob.delStatus("corporeal")
+		src.wraith_mob.set_loc(T)
+		boutput(T, "<span class='bold' style='color:red;font-size:150%'>You have assumed control of this body! You don't have long...</span>")
+		RegisterSignal(T, COMSIG_MOB_DEATH, PROC_REF(return_wraith))
+		APPLY_ATOM_PROPERTY(T, PROP_MOB_NO_SELF_HARM, T)
+		message_ghosts("<b>[src.wraith_mob]</b> has possessed <b>[src.owner]</b> at [log_loc(src.owner, ghostjump = TRUE)]!")
+		for (var/mob/dead/target_observer/TO in src.wraith_mob.observers)
+			if (TO == src.human_observer)
+				continue
+			TO.set_observe_target(src.owner)
+
+	proc/return_wraith()
+		UnregisterSignal(src.owner, COMSIG_MOB_DEATH)
+		if (QDELETED(src.wraith_mob) || src.wraith_mob.mind == src.wraith_mind)
 			return
+		src.wraith_mob.set_loc(get_turf(src.owner))
+		src.wraith_mob.set_dir(src.owner.dir)
```

Comment:
ye! it's just an aesthetic choice -- it felt neat to have them emerge in the exact same location and direction of the mob they were possessing, 'cuz I thought it would be a fun allusion to them puppeting the person's actions